amida-tech/amida-auth-microservice
# Feature/ser 221 auth create admin on sync

If this PR fixes a bug, you _must_ add test cases representative of the bug.
#### What's this PR do?
Will seed the auth database with a basic config admin user. We can change this with environmental variables later, but Mike thinks this is not entirely important right now.

This PR will:

1. Seed a new admin user upon start up only if the `user` database is entirely empty.
2. Generate a strong password by inserting a random upper case letter, a lower case letter, a number and a symbol, at random points of a generated UUID.
3. Log output the newly created password using Winston.
4. Will now be tested in the auth integrations.

#### Related JIRA tickets:

#### How should this be manually tested?
Run the tests using `yarn test`. Please keep in mind that because the seeding happens first thing, it wipes out all users in your current `users` database. This is doing _no more damage_ than the tests were before. We need to create a new testing database to preserve data, and I think @mountHouli has an eye on that. 

For manual testing, check your `user` database for existing users.

When users exist, turn on the application using `yarn start` while you have existing users in there. Confirm that no additional users are added. 

When no users exist, turn on the application using `yarn start`. A new user should be created, and your log output should mention this as well as provide the password to access this user. 

#### Any background context you want to provide?
#### Screenshots (if appropriate):
